corehq/apps/data_dictionary/util.py

```diff
@@ -96,7 +96,8 @@ def create_properties_for_case_types(domain, case_type_to_prop):
         try:
             case_type_obj = current_case_types[case_type]
         except KeyError:
-            case_type_obj = CaseType.objects.create(domain=domain, name=case_type)
+            case_type_obj = CaseType(domain=domain, name=case_type)
+            case_type_obj.save()  # save() method clears cache
```

Comment:
It seems like we would want to use this pattern anywhere `CaseType` is created, but other developers may opt for `.objects.create()` not knowing they should do this instead. For example, should it be done here too?

https://github.com/dimagi/commcare-hq/blob/01aabf391f035c183233da549948d96a1b0ac7fd/corehq/apps/data_dictionary/models.py#L37

What do you think of overriding the `create` model manager method (would require adding a custom manager to `CaseType`) to make it clear the cache?

Reply:
I like that idea. It'll also give me the opportunity to move `CaseType.get_or_create()` to `CaseType.objects.get_or_create()`.

Reply:
Oh! My mistake. It turns out that `Foo.objects.create()` [calls `foo.save()` already](https://github.com/django/django/blob/a7b0e50eadba8f0420013605c70eb790280b0fd2/django/db/models/query.py#L658).

I'm going to close this PR, and come back to move the `get_or_create()` class methods (`CaseProperty` has one too) another day.